src/utils.rs

```diff
@@ -0,0 +1,114 @@
+use blake2::digest::{Update, VariableOutput};
+use blake2::Blake2bVar;
+use std::cmp::min;
+
+// Oracles
+
+/// Takes as input a hash and inverse probability $np$ and returns a
+/// uniformly distributed integer in [1, np] (c.f. Appendix B, Alba paper).
+pub fn oracle_uniform(hash: &[u8], n: usize) -> usize {
+    // Computes the integer reprensation of hash* modulo n when n is not a
+    // power of two. *(up to 8 bytes, in little endian)
+    fn mod_non_power_of_2(hash: &[u8], n: usize) -> usize {
+        fn log_base2(x: usize) -> usize {
+            usize::BITS as usize - x.leading_zeros() as usize - 1
+        }
+        let epsilon_fail: usize = 1 << 40; // roughly 1 trillion
```

Comment:
One trillion seems low compared to typical security / reliability parameters. Is this going to be improved in the future?

Reply:
I left the code as is (I think that was from Arnaud) as I was not sure what to choose here as parameter. I think the best would be to take as input lambda and use this.